src/main/kotlin/lotto/domain/Prize.kt

```diff
@@ -3,16 +3,20 @@ package lotto.domain
 enum class Prize(
     val value: Int,
     val matchCount: Int,
-    private val matcher: (Int) -> Boolean,
+    private val matcher: (Int, Boolean) -> Boolean,
 ) {
-    FIRST(2_000_000_000, 6, { it == 6 }),
-    SECOND(1_500_000, 5, { it == 5 }),
-    THIRD(50_000, 4, { it == 4 }),
-    FOURTH(5_000, 3, { it == 3 }),
-    NONE(0, 0, { false }),
+    FIRST(2_000_000_000, 6, { count, _ -> count == 6 }),
```

Comment:
I feel uncomfortable using `_` in my codes, but as also seen on the lectures from Jason and library codes on Kotlin,
it looks like it is common. I try to differentiate variable names but it's hard.
For example
```
class Car(_position: Int) {
    val position = _position
}
```

Is it something I should get used to when using Kotlin?

Reply:
I feel very related 😆
This is quite common in Kotlin actually, particularly when you are using backing properties. (btw, if you were an Android developer, you probably used backing properties in observable types a lot)

https://kotlinlang.org/docs/properties.html#backing-properties
https://developer.android.com/kotlin/flow/stateflow-and-sharedflow#stateflow